python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -720,7 +720,10 @@ def _get_self_actor_id(self) -> str:
         if self._read_by_adag_driver:
             # The reader is the driver process.
             # Use the actor ID of the DAGDriverProxyActor.
-            assert len(self._reader_and_node_list) == 1
+            assert len(self._reader_and_node_list) > 0
+            # The first element of the reader_and_node_list is always the
+            # proxy actor if it's a reader per the creation of reader_and_node_list
+            # in CompiledDAGNode.
             driver_actor = self._reader_and_node_list[0][0]
```

Comment:
inserting proxy actor at 0 index during creation as mentioned in comment above right now, would it be a better idea to just pass it through to the channel instead of relying on this logic?